include/seqan3/range/views/minimiser.hpp

```diff
@@ -407,26 +506,61 @@ struct minimiser_fn
         return adaptor_from_functor{*this, window_values_size};
     }
 
-    /*!\brief                           Call the view's constructor with two arguments: the underlying view and an
-     *                                  integer indicating how many values one window contains.
-     * \param[in] urange                The input range to process. Must model std::ranges::viewable_range and
-     *                                  std::ranges::forward_range.
-     * \param[in] window_values_size    The number of values in one window.
-     * \returns                         A range of converted values.
+    /*!\brief Call the view's constructor with two arguments: the underlying view and an integer indicating how many
+     *        values one window contains.
+     * \param[in] urange1 The input range to process. Must model std::ranges::viewable_range and
+     *                    std::ranges::forward_range.
+     * \param[in] window_values_size The number of values in one window.
+     * \returns  A range of converted values.
      */
-    template <std::ranges::range urng_t>
-    constexpr auto operator()(urng_t && urange, size_t const window_values_size) const
+    template <std::ranges::range urng1_t>
+    constexpr auto operator()(urng1_t && urange1, size_t const window_values_size) const
     {
-        static_assert(std::ranges::viewable_range<urng_t>,
-            "The range parameter to views::minimiser cannot be a temporary of a non-view range.");
-        static_assert(std::ranges::forward_range<urng_t>,
-            "The range parameter to views::minimiser must model std::ranges::forward_range.");
+        static_assert(std::ranges::viewable_range<urng1_t>,
+                      "The range parameter to views::minimiser cannot be a temporary of a non-view range.");
+        static_assert(std::ranges::forward_range<urng1_t>,
+                      "The range parameter to views::minimiser must model std::ranges::forward_range.");
 
-        if (window_values_size == 1)
-            throw std::invalid_argument{"The chosen window_valzes_size is not valid. "
-                                        "Please choose a value greater than 1."};
+        if (window_values_size == 1) // Would just return urange1 without any changes
+            throw std::invalid_argument{"The chosen window_values_size is not valid. "
+                                        "Please choose a value greater than 1 or use two ranges."};
 
-        return minimiser_view{urange, window_values_size};
+        return minimiser_view{urange1, window_values_size};
+    }
+
+
+    //!\brief Store the number of values in one window and the second range and return a range adaptor closure object.
+    template <std::ranges::range urng2_t>
+    constexpr auto operator()(size_t const window_values_size, urng2_t && urange2) const
```

Comment:
I would tend to

```
range1 | minimiser(range2, 4);
```

Reply:
I find the current implementation more intuitive, because usually when you have variables, where you can choose if you set them, there appear in the end of the variable list.

Reply:
I would also tend to 

```c++
range1 | minimiser(range2, 4);
```

Let's wait what the second reviewer thinks.

Reply:
I would prefer this order as well. The adaptor is anyway returning different types depending on the input. This design feels more appropriate since the ranges are closer in the call.